src/Features/CSharp/Portable/ExtractMethod/CSharpExtractMethodService.cs

```diff
@@ -15,10 +15,10 @@ namespace Microsoft.CodeAnalysis.CSharp.ExtractMethod;
 [ExportLanguageService(typeof(IExtractMethodService), LanguageNames.CSharp)]
 [method: ImportingConstructor]
 [method: SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
-internal sealed class CSharpExtractMethodService() : AbstractExtractMethodService<
-    CSharpSelectionValidator,
-    CSharpMethodExtractor,
-    CSharpSelectionResult,
+internal sealed partial class CSharpExtractMethodService() : AbstractExtractMethodService<
+    CSharpExtractMethodService.CSharpSelectionValidator,
+    CSharpExtractMethodService.CSharpMethodExtractor,
+    CSharpExtractMethodService.CSharpSelectionResult,
```

Comment:
core helper types became nested types of the service.  this way they do not have to repeat type arguments.